pkg/allocator/allocator.go

```diff
@@ -716,6 +745,55 @@ func (a *Allocator) Allocate(ctx context.Context, key AllocatorKey) (idpool.ID,
 	return 0, false, false, err
 }
 
+func (a *Allocator) GetWithRetry(ctx context.Context, key AllocatorKey) (idpool.ID, error) {
+	getID := func() (idpool.ID, error) {
+		id, err := a.Get(ctx, key)
+		if err != nil {
+			return 0, err
+		}
+
+		if id == idpool.NoID {
+			return 0, fmt.Errorf("security identity not found for key %s", key.String())
+		}
+
+		return id, nil
+	}
+
+	// Make a copy of the template and customize it.
+	boff := a.backoffTemplate
+	boff.Name = key.String()
+
+	var id idpool.ID
+	var err error
+
+	for attempt := 0; attempt < a.maxAllocAttempts; attempt++ {
+		id, err = getID()
```

Comment:
I can move the `id idpool.ID` inside the for statement scope, but if I do as suggested, it will use variable shadowing, which means that the error on declared before the for statement will get no assignment from within the statement, because `:=` is used.

Ref:
https://stackoverflow.com/questions/76143322/golang-shadowing-variable
https://go.dev/play/p/DWhU5CB5Eqo